### PR TITLE
catches errors in case

### PR DESCRIPTION
**Pull request by elliottetzkorn**
@rfroetscher

### CODE REVIEW

File: app/models/shout.rb [hunk 1]
```diff
@@ -39,39 +40,58 @@ def self.make_from_handles(options = {})
   end
 
   def sender_handle
-    sender.handle
+    if sender
```
Comment (dollschasingmen) [inline app/models/shout.rb]: should use http://apidock.com/rails/Object/try

File: app/models/shout.rb [hunk 2]
```diff
@@ -39,39 +40,58 @@ def self.make_from_handles(options = {})
...
-    user_hash = SlackInfoRetriever.info_for_member(sender.handle)
-    if !user_hash.empty?
-      user_hash["profile"]["image_original"]
+    if sender
+      user_hash = SlackInfoRetriever.info_for_member(sender.handle)
+      if !user_hash.empty?
+        user_hash["profile"]["image_original"]
+      else
+        ""
+      end
```
Comment (dollschasingmen) [inline app/models/shout.rb]: what do you think about this? (also ruby style says that you should use single qoutes for string literals unless you need to do interpolation or escaping)

    slack_info = SlackInfoRetriever.info_for_member(sender_handle)
    slack_info.present? ? slack_info['profile']['image_original'] : ''

File: app/models/shout.rb [hunk 3]
```diff
@@ -39,39 +40,58 @@ def self.make_from_handles(options = {})
...
       ""
     end
   end
 
   def receiver_avatar
-    user_hash = SlackInfoRetriever.info_for_member(receiver.handle)
-    if !user_hash.empty?
-      user_hash["profile"]["image_original"]
+    if receiver
+      user_hash = SlackInfoRetriever.info_for_member(receiver.handle)
```
Comment (dollschasingmen) [inline app/models/shout.rb]: we seem to be duplicating a lot of logic here i wonder if we can just have a method like

     def fetch_stack_info(type)
           ....
     end

where type can be like :receiver or :sender 

it returns a a slack_info and then you can be all like `slack_info[k1][k2]`

Comment (rfroetscher) [inline app/models/shout.rb]: :+1: for this idea

File: app/models/shout.rb (hunk 1 above)
Comment (rfroetscher) [inline app/models/shout.rb]: we dont allow unpersisted users to be saved anymore, so this really shouldnt be an issue

**Pull request closed without merging**